commands/publish_dafsa.py

```diff
@@ -0,0 +1,124 @@
+            print(f"Other error occurred: {err}")
+
+    return inner
+
+
+@handle_request_errors
+def get_latest_hash():
+    response = requests.get(COMMIT_HASH_URL)
+    response.raise_for_status()
+    return response.json()[0]["sha"]
+
+
+@handle_request_errors
+def download_resources(directory, *urls):
+    for url in urls:
+        # file_location is found by appending the file_name(at the end of url string) to temp directory
+        file_name = os.path.basename(url)
+        file_location = os.path.join(directory, file_name)
+        print(file_location)
+        response = requests.get(url, stream=True)
+        response.raise_for_status()
+
+        with open(file_location, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    f.write(chunk)
+                else:
+                    print("Error!!!")
+
+
+def make_dafsa_and_publish(latest_hash):
+    with tempfile.TemporaryDirectory() as tmp:
+        download_resources(tmp, LIST_URL, MAKE_DAFSA_PY, PREPARE_TLDS_PY)
+        """
+        prepare_tlds.py is called with the two arguments the location of 
+        the downloaded public suffix list and the name of the output file
+        """
+        output_binary_name = "etld_data.json"
+
+        prepare_tlds_py_path = os.path.join(tmp, "prepare_tlds.py")
+        raw_psl_path = os.path.join(tmp, "public_suffix_list.dat")
+        output_binary_path = os.path.join(tmp, output_binary_name)
+
+        # Make the DAFSA
+        run = subprocess.run(
+            ["python3", prepare_tlds_py_path, raw_psl_path, output_binary_path]
+        )
+        if run.returncode != 0:
+            print("DAFSA Build Failed !!!")
+            return 1
+
+        subprocess.run(["ls", tmp])
+        client = Client(
+            server_url=SERVER, auth=AUTH, bucket=BUCKET_ID, collection=COLLECTION_ID
+        )
+        # Upload the attachment
+        mimetype = "application/octet-stream"
+        filename = output_binary_name
+        filecontent = open(output_binary_path, "rb").read()
+        record_uri = client.get_endpoint("record", id=RECORD_ID)
+        attachment_uri = f"{record_uri}/attachment"
+        multipart = [("attachment", (filename, filecontent, mimetype))]
+        commit_hash = json.dumps({"commit-hash": latest_hash})
+
+        body, _ = client.session.request(
+            method="post", data=commit_hash, endpoint=attachment_uri, files=multipart
+        )
+        print(body)
+
+
+def publish_dafsa():
+    # client = Client(
+    #     server_url=SERVER, auth=AUTH, bucket=BUCKET_ID, collection=COLLECTION_ID
+    # )
+    latest_hash = get_latest_hash()
+    # try:
+    # record = client.get_record(id=RECORD_ID)
+    # print(record)
+    # except KintoException as e:
+    #     print(e)
+
+    # if record["data"]["latest-commit-hash"] == latest_hash:
+    #     return 1
+    # else:
+    make_dafsa_and_publish(latest_hash)
+
+
+publish_dafsa()
```

Comment:
We don't want to execute this on import.

The command should be ran like this `python aws_lambda.py publish_dafsa`